plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -275,11 +286,41 @@ func (s *SQLServer) Gather(acc telegraf.Accumulator) error {
 }
 
 func (s *SQLServer) gatherServer(server string, query Query, acc telegraf.Accumulator) error {
-	// deferred opening
-	conn, err := sql.Open("mssql", server)
-	if err != nil {
-		return err
+	// connection object
+	var conn *sql.DB
+
+	// setup connection based on authentication
+	rx := regexp.MustCompile(`(?i)(^|[;])(?:(\s*Authentication=Active Directory Integrated\s*))([;]|$)`)
+
+	// when auth mechanism is mentioned as AAD in the connection string, use managed-identity token
+	if rx.MatchString(server) {
```

Comment:
Is "Authentication=Active Directory Integration" property a mandatory field in connection string? If that property is missing, and password is not specified, connection would still work right?

Reply:
Yes, for AAD auth the connection string needs to include "Authentication=Active Directory Integration" property (I'm referring to [this ](https://docs.microsoft.com/en-us/azure/azure-sql/database/authentication-aad-configure?tabs=azure-powershell#active-directory-integrated-authentication-1) doc). With the current code, if the property is not included it'll be processed as SQL Auth, and for your example, it won't connect since there are no credentials.

Reply:
I think that is AAD and different from MSI auth. Can you help remove the property Authentication from your connection string and see if you're able to connect in your test systems. My understanding is that this is not a required field and connection should succeed. Please help verify.

Reply:
Thank you @masree for highlighting and clarifying this. Yes, MSI auth does not require the Authentication property in the connection string. I have reverted the code to use existence of "Password" as the definition of whether to use SQL Auth or AAD/MSI Auth. Readme file has been correspondingly modified as well.